### ISSUE TITLE
Panic on missing config file

### ISSUE THREAD

**Issue by ruflin** [elastic/libbeat#75]
Currently if the path to the config file does not exist, all beats crash. A proper output should be given that explains the problem and the app should be shut down. Here the error message from filebeat:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x61927]

goroutine 1 [running]:
github.com/elastic/libbeat/beat.(*Beat).LoadConfig(0xc820014a00)
        /.../gopath/src/github.com/elastic/filebeat/Godeps/_workspace/src/github.com/elastic/libbeat/beat/beat.go:81 +0x437
main.main()
        /Users/ruflin/Dev/gopath/src/github.com/elastic/filebeat/main.go:34 +0x1aa

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/local/Cellar/go/1.5/libexec/src/runtime/asm_amd64.s:1696 +0x1

goroutine 5 [syscall]:
os/signal.loop()
        /usr/local/Cellar/go/1.5/libexec/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
        /usr/local/Cellar/go/1.5/libexec/src/os/signal/signal_unix.go:28 +0x37
make: *** [run] Error 2
```

**Comment by tsg** [comment]
Seems to be filebeat specific, topbeat behaves:

    ./topbeat -e -c etc/filebeat.ym -d "publish"
    Failed to read etc/filebeat.ym: open etc/filebeat.ym: no such file or directory. Exit.

**Comment by tsg** [comment]
But I think there is something which we do quite bad in the Beater code in libbeat, we're inconsistent about how we handle errors and this is causing issues like this one and also #58. The general rules I'd make:

 * If a function is returning error (has error as the return type), that should be its only error reporting mechanism. It shouldn't log the error or exit the process.
 * The function caller can either return the error on its turn (perhaps enhanced with more information), or log it and take the required action (e.g. exit the process). If it does exit, it must have no return type and the doc comment should say that it can exit the process.

If there are no objections @ruflin @monicasarbu @urso @andrewkroh , I can create a PR to update the logic in the Beater code and all beats.

**Comment by andrewkroh** [comment]
@tsg No objections here. SGTM

**Comment by tsg** [comment]
Opened #77 with the proposal. I'll follow up with the individual Beats adjustments.

**Comment by ruflin** [comment]
@tsg Interesting. I get the same error for topbeat and packetbeat on master:

```
./topbeat -c wer
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4066c67]

goroutine 1 [running]:
github.com/elastic/libbeat/beat.(*Beat).LoadConfig(0xc820012840)
	/.../gopath/src/github.com/elastic/topbeat/Godeps/_workspace/src/github.com/elastic/libbeat/beat/beat.go:81 +0x437
main.main()
	/.../gopath/src/github.com/elastic/topbeat/main.go:19 +0x1b1

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/Cellar/go/1.5/libexec/src/runtime/asm_amd64.s:1696 +0x1

goroutine 5 [syscall]:
os/signal.loop()
	/usr/local/Cellar/go/1.5/libexec/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/Cellar/go/1.5/libexec/src/os/signal/signal_unix.go:28 +0x37

**Comment by ruflin** [comment]
@tsg I agree with the convention. I would event suggest to have an "exit" function in libbeat that should be used. By calling this function, even some cleanup could be done if necessary and we make sure always the same "exit" method is used.

**Comment by tsg** [comment]
@ruflin, ah, yes, I was wrong in my testing before. This was fixed as part of #58 but godeps weren't updated in Beats since then but I had the topbeat and packetbeat binaries still built against the fixed version.

**Comment by urso** [comment]
+1
but next to returning an error have a log message on debug level I would say. Something to trace the error it's origin.
